internal/ethapi/api.go

```diff
@@ -1048,13 +1048,10 @@ func DoCall(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash
 			SkipAccountChecks: true,
 		}
 
-		_, result, finalize, err := runMEVM(ctx, b, state, header, tx, msg, true)
+		_, result, _, err := runMEVM(ctx, b, state, header, tx, msg, true)
```

Comment:
Estimate gas (and any other call) should not call finalize on the confidential store. Otherwise, the values are written for calls. This was not touched before on-purpose since we needed for the forge integration. Now it is not required anymore.

Reply:
Ditto.  Consider adding this as a code comment.